### PR TITLE
Fix: Replaced explanation of option object; corrected first example f…

### LINKED ISSUE #12874: new cap: option capIsNewExceptionPattern is not behaving as documented
Documentation for `capIsNewExceptionPattern` says the following:
> allows any uppercase-started **function names that match** the specified regex pattern to be called without the new operator.

However, the actual behaviour seems to be that entire [call expression is matched](https://github.com/eslint/eslint/blob/439c83342c364ba3ce5168d54e165b1fe3e35630/lib/rules/new-cap.js#L211) against the pattern, not just the function name.

Either `capIsNewExceptionPattern` should match only function name, or the documentation should reflect the actual behaviour.

I believe matching code with regex is fundamentally wrong and the first option is more sane solution. So, the documentation should stay as is, behaviour corrected and example updated, but that might be a breaking change.

Additionally, [the example](https://eslint.org/docs/rules/new-cap#capisnewexceptionpattern) given as "correct code for this rule" is actually producing warnings (see demo), the reason being the regex configured does not match anything in the code at all.

Also other options of this rule may be poorly documented as well, for example `capIsNewExceptions` also states that it matches names, while actually it can match either name or full expression.

**Tell us about your environment**

Node version: v12.10.0
npm version: v6.10.3
Local ESLint version: v6.8.0 (Currently used)
Global ESLint version: Not found

Also same result on whatever version is currently deployed as demo.

**Please show your full configuration:**

<details>
<summary>Demo links</summary>

Expected no warnings: https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbmV3LWNhcDogW1wiZXJyb3JcIiwgeyBcImNhcElzTmV3RXhjZXB0aW9uUGF0dGVyblwiOiBcIl5CYXIkXCIgfV0qL1xuXG5mb28uQmFyKCk7Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMCwic291cmNlVHlwZSI6Im1vZHVsZSIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==

Not expected to ignore:
https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbmV3LWNhcDogW1wiZXJyb3JcIiwgeyBcImNhcElzTmV3RXhjZXB0aW9uUGF0dGVyblwiOiBcIl5mb29cXC5CYXIkXCIgfV0qL1xuXG5mb28uQmFyKCk7Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMCwic291cmNlVHlwZSI6Im1vZHVsZSIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==

Documentation example, wrong on several accounts: https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbmV3LWNhcDogW1wiZXJyb3JcIiwgeyBcImNhcElzTmV3RXhjZXB0aW9uUGF0dGVyblwiOiBcIl5QZXJzb25cXC4uXCIgfV0qL1xuXG52YXIgZnJpZW5kID0gcGVyc29uLkFjcXVhaW50YW5jZSgpO1xudmFyIGJlc3RGcmllbmQgPSBwZXJzb24uRnJpZW5kKCk7Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMCwic291cmNlVHlwZSI6Im1vZHVsZSIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==
</details>

**What did you do?**

Configuration:
```
eslint new-cap: ["error", { "capIsNewExceptionPattern": "^Bar$" }]
```

Code:
```js
foo.Bar();
```

**What did you expect to happen?**

No warnings as function named `Bar` should be ignored.

**What actually happened? Please include the actual, raw output from ESLint.**

A warning against function `Bar`.

**Are you willing to submit a pull request to fix this bug?**

Only if it's as trivial as `pattern.test(calleeName)`, but then it's probably trivial for anyone else as well.

### PR DESCRIPTION
**Pull request by jordanmoore753**
…or capIsNewExceptionPattern; included additional correct case (fixes #12874)

#### Prerequisites checklist

- [X ] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).
- [x] The team has reached consensus on the changes proposed in this pull request. If not, I understand that the evaluation process will begin with this pull request and won't be merged until the team has reached consensus.

#### What is the purpose of this pull request? (put an "X" next to an item)

[X ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

I corrected the first **correct** example for the `capIsNewExceptionPattern` section, in addition to adding a second **correct** example. I also updated the **Options** section at the top with a more descriptive summary of what the `option` object can be used for.

### CODE REVIEW

File: docs/rules/new-cap.md [hunk 1]
```diff
@@ -135,14 +135,21 @@ function foo(arg) {
 
 ### capIsNewExceptionPattern
 
-Examples of additional **correct** code for this rule with the `{ "capIsNewExceptionPattern": "^Person\.." }` option:
+Examples of additional **correct** code for this rule with the `{ "capIsNewExceptionPattern": "^person\.." }` option:
```
Comment (mdjermanovic) [inline docs/rules/new-cap.md]: :+1:

File: docs/rules/new-cap.md [hunk 2]
```diff
@@ -135,14 +135,21 @@ function foo(arg) {
...
 
 var friend = person.Acquaintance();
 var bestFriend = person.Friend();
 ```
+Examples of additional **correct** code for this rule with the `{ "capIsNewExceptionPattern": "\.Bar$" }` option:
+
+```js
+/*eslint new-cap: ["error", { "capIsNewExceptionPattern": "\.Bar$" }]*/
+
+foo.Bar(); // Lint-Free!
```
Comment (mdjermanovic) [inline docs/rules/new-cap.md]: There is no need for `// Lint-Free!`, this is already an example of correct code.

Also, can we please add a similar example for `newIsCapExceptionPattern`?

**Comment by nzakas** [comment]
@jordanmoore753 are you still working on this?

**Comment by jordanmoore753** [comment]
Hi @nzakas! Yes I'm still on this, sorry for the delay. I've had a lot of work lately and haven't had the chance to address this. I'll work on it this week.

**Comment by nzakas** [comment]
No worries, the world is crazy at the moment. Just didn’t want to hand this off if you were still going to work on it.

**Comment by nzakas** [comment]
@jordanmoore753 just one more check to see if you're still working on this. We completely understand if you can't finish this up, we'd just like to know if we should get someone else to finish this up. Thanks!

**Comment by mdjermanovic** [comment]
Closing due to inactivity and in favor of https://github.com/eslint/eslint/pull/14725.

**Pull request closed without merging**